pkg/resourcemanager/azuresql/azuresqlmanageduser/azuresqlmanageduser_reconcile.go

```diff
@@ -169,7 +170,6 @@ func (s *AzureSqlManagedUserManager) Delete(ctx context.Context, obj runtime.Obj
 	db, err := s.ConnectToSqlDbAsCurrentUser(ctx, instance.Spec.Server, instance.Spec.DbName)
 	if err != nil {
 		instance.Status.Message = errhelp.StripErrorIDs(err)
-		instance.Status.Provisioning = false
```

Comment:
Why does this one go away? Is it just because the instance is going away anyway? Or just because Provisioning would already be false by this point?

Reply:
There are a few cases:
1. Provisioning was set to false already because the resource was successfully created already. In this case, since we're just deleting, it doesn't make sense to set provisioning back to true (and we don't).
2. Provisioning was true, which means the resource wasn't successfully created yet. In this case it might make sense to set it to false since we're about to delete it, but doing so seems somewhat pointless because we're about to delete the resource. Certainly in this case, (the error case), it doesn't make sense to set provisioning to false. It might might sense to do it as a default at the beginning of delete.

Based on my discussions with Erin, provisioning was originally intended to actually reflect _when we are doing a deployment operation to Azure_. I think longer term we should revisit this and possibly just remove the notion of informing the user of that aspect anyway. We should just tell them:
```
Reconciling
Steady
ReconcileFailed
```

I think this more clearly covers all of the cases and allows us to set `Reconciling` for cases like `Delete` where we really aren't `provisioning` but we are doing _something_.

Reply:
Makes sense, thanks!